### PR TITLE
Fixed InputNorm to correctly normalize inputs

### PR DESCRIPTION
**Pull request by Theomat**
Also fixed mypy issues

### CODE REVIEW

**Comment by cswinter** [comment]
Thanks! Could you elaborate on the changes to the algorithm? I originally based this code on https://www.johndcook.com/blog/standard_deviation/ (we should add this link back in, got lost when copied the code) and the numerics are quite subtle, so I have a hard time telling if the new formulation is correct. I think we'll want either a good theoretical justification for why it works, or empirically determine its accuracy on large inputs (probably the second one).

File: ppo/train.py [hunk 1]
```diff
@@ -175,7 +173,7 @@ def __init__(
         self.embedding = nn.ModuleDict(
             {
                 name: nn.Sequential(
-                    InputNorm(len(entity.features)),
+                    InputNorm(len(entity.features), cliprange=1),
```
Comment (cswinter) [inline ppo/train.py:176]: I think this is too aggressive, it will clip 32% of normally distributed values which could throw away a lot of information (and could clip even much more for non-normally distributed input features). The original default of 5 wasn't chosen in a super principled way (I just cargo-culted it from OpenAI Five), so there might be a better value, but it still seems about right.

**Review by cswinter — CHANGES_REQUESTED** [review]

File: ppo/train.py (hunk 1 above)
Comment (Theomat) [reply ppo/train.py:176]: Yes this is something I just wanted to test, this should be removed, my bad.

Comment (cswinter) [reply ppo/train.py:176]: E.g. 42% of uniformly distributed values would get clipped, and this would be even worse for more skewed distributions:
```
>>> import numpy as np
>>> np.random.uniform(-1.73, 1.73, size=(100000,)).std()
0.9988141429161412
>>> (0.73 * 2) / (1.73 * 2)
0.42196531791907516
```

**Comment by Theomat** [comment]
> Thanks! Could you elaborate on the changes to the algorithm? I originally based this code on https://www.johndcook.com/blog/standard_deviation/ (we should add this link back in, got lost when copied the code) and the numerics are quite subtle, so I have a hard time telling if the new formulation is correct. I think we'll want either a good theoretical justification for why it works, or empirically determine its accuracy on large inputs (probably the second one).

I didn't know about such methods, I believe that instead of the simple derivation I made, the methods you linked [Welford's method] should be used, see John D. Cook's other [article on numerical precision](https://www.johndcook.com/blog/2008/09/26/comparing-three-methods-of-computing-standard-deviation/).

**Comment by cswinter** [comment]
Yeah, no worries, it's a pretty tricky piece of code, I appreciate the initiative 😄
